inc/StyleUtils.php

```diff
@@ -4,101 +4,161 @@
-                        continue;
-                    }
-                }
-                $stylesheets[$mode][$incbase . $file] = $webbase;
+    /**
+     * @return array
+     */
+    protected function mergeStyleInis() {
+        static $combined = [];
+        if (empty($combined) || $this->reinit) {
+            global $conf;
+            global $config_cascade;
+            $stylesheets = array(); // mode, file => base
+            $incbase = tpl_incdir($this->tpl);
+
+            // guaranteed placeholder => value
+            $replacements = array(
+                '__text__' => "#000",
+                '__background__' => "#fff",
+                '__text_alt__' => "#999",
+                '__background_alt__' => "#eee",
+                '__text_neu__' => "#666",
+                '__background_neu__' => "#ddd",
+                '__border__' => "#ccc",
+                '__highlight__' => "#ff9",
+                '__link__' => "#00f",
+            );
+
+            // merge all styles from config cascade
+            if (!is_array($config_cascade['styleini'])) {
+                trigger_error('Missing config cascade for styleini',E_USER_WARNING);
             }
 
-            // replacements
-            if(is_array($data['replacements'])){
-                $replacements = array_merge($replacements, $this->cssFixreplacementurls($data['replacements'],$webbase));
+            // allow replacement overwrites in preview mode
+            if($this->preview) {
+                $config_cascade['styleini']['local'][] = $conf['cachedir'].'/preview.ini';
             }
-        }
 
-        // load configs's style.ini
-        $webbase = DOKU_BASE;
-        $ini = DOKU_CONF."tpl/$tpl/style.ini";
-        $incbase = dirname($ini).'/';
-        if(file_exists($ini)){
-            $data = parse_ini_file($ini, true);
+            foreach (array('default','local','protected') as $config_group) {
+                if (empty($config_cascade['styleini'][$config_group])) continue;
 
-            // stylesheets
-            if(isset($data['stylesheets']) && is_array($data['stylesheets'])) foreach($data['stylesheets'] as $file => $mode){
-                $stylesheets[$mode][$incbase.$file] = $webbase;
-            }
+                // set proper server dirs
+                $webbase = $this->getServerPath($config_group);
+
+                foreach ($config_cascade['styleini'][$config_group] as $file) {
+                    // replace the placeholder with the name of the current template
+                    $file = str_replace('%TEMPLATE%', $this->tpl, $file);
 
-            // replacements
-            if(isset($data['replacements']) && is_array($data['replacements'])){
-                $replacements = array_merge($replacements, $this->cssFixreplacementurls($data['replacements'],$webbase));
+                    if (file_exists($file)) {
+                        $config = call_user_func_array('parse_ini_file', array_merge(array($file), array(true)));
```

Comment:
Why the `call_user_func_array()` here? Is there a reason why `parse_ini_file` can't be called directly?